autopwn.py

```diff
@@ -299,8 +299,7 @@ def InitArgsConf():
 
 def InitArgsScanType():
     scantype = ScanType.Ping
-
-    if args.scantype == "arp" or args.scantype == None:
+    if args.scantype == "arp":
         if is_root():
```

Comment:
it is the same with the old conditions, just reduced, since the old conditions:

```python
    ...
    if args.scantype == "arp":
        if not is_root():
            error("You must be root to do an arp scan!")
            scantype = ScanType.Ping
        else:
            scantype = ScanType.Arp
    elif args.scantype == "ping":
        scantype = ScanType.Ping
    elif args.scantype == "" or type(args.scantype) == None or args.scantype == None:
        if is_root():
            scantype = ScanType.ARP
        else:
            scantype = ScanType.Ping
```

the old conditions will also land in the end to the ping scan type if the user did not executed it in root or if the user did not selected arp, so the revision defaulted to ping scan type, and will later evaluated if it matches the condition for arp scan (the user should have selected arp, and is root), if it is it will use arp scan, if not, then ping scan. although I addressed it again in the last commit, 9754b6bc54c381d2457977e1425228d9f76eaa4c:

> use ping scan if:
> - there is no selected scan and the user is not root
> - or the user selected ping scan
>
> default to arp scan if:
> - there is no selected scan and the user is root
> - or the user selected ping scan